optaplanner-examples/src/main/java/org/optaplanner/examples/nurserostering/domain/solver/MovableShiftAssignmentSelectionFilter.java

```diff
@@ -17,16 +17,19 @@
 package org.optaplanner.examples.nurserostering.domain.solver;
 
 import org.optaplanner.core.api.domain.entity.PinningFilter;
+import org.optaplanner.core.impl.heuristic.selector.common.decorator.SelectionFilter;
+import org.optaplanner.core.impl.score.director.ScoreDirector;
 import org.optaplanner.examples.nurserostering.domain.NurseRoster;
 import org.optaplanner.examples.nurserostering.domain.ShiftAssignment;
-import org.optaplanner.examples.nurserostering.domain.ShiftDate;
 
-public class MovableShiftAssignmentSelectionFilter implements PinningFilter<NurseRoster, ShiftAssignment> {
+public class MovableShiftAssignmentSelectionFilter implements SelectionFilter<NurseRoster, ShiftAssignment> {
```

Comment:
I have to say... I don't like the fact that, in order to filter moves in selectors, users need to implement this interface.
Ideally, I'd like `ScoreDirector` to be an implementation detail. Yet, it keeps leaking.

Reply:
Yea, sounds like maybe SelectionFilter, instead of creating a new type of Filter like PinningFilter that doesn't work on moves (and shouldn't due to naming).